plugin-server/src/main/ingestion-queues/session-recording-v2/sessions/session-console-log-store.ts

```diff
@@ -27,25 +35,55 @@ export class SessionConsoleLogStore {
 
     public async storeSessionConsoleLogs(logs: ConsoleLogEntry[]): Promise<void> {
         if (logs.length === 0 || !this.topic) {
-            return
+            return Promise.resolve()
         }
 
-        await this.producer.queueMessages({
-            topic: this.topic,
-            messages: logs.map((log) => ({
-                value: JSON.stringify(log),
-                key: log.log_source_id, // Using session_id as the key for partitioning
-            })),
-        })
+        if (this.pendingPromises.length >= this.promiseLimit) {
+            this.syncPromise = this.sync()
```

Comment:
I feel like this _could_ be more manageable if you here `.finally(() => { this.syncPromise = null })` - that way the promise is self-cleaning. Not 100% sure but I also feel like you need to check for the syncPromise first? Otherwise you could end up overwriting an existing one?